src/Translation.jl

```diff
@@ -0,0 +1,345 @@
+            Translation{N, VN, S}(X, v)
+    end
+end
+
+# constructor from a Translation: perform the translation immediately
+Translation(tr::Translation{N}, v::AbstractVector{N}) where {N<:Real} =
+    Translation(tr.X, tr.v + v)
+
+"""
+    +(X::LazySet, v::AbstractVector)
+
+Convenience constructor for a translation.
+
+### Input
+
+- `X` -- convex set
+- `v` -- vector
+
+### Output
+
+The symbolic translation of ``X`` along vector ``v``.
+"""
++(X::LazySet, v::AbstractVector) = Translation(X, v)
+
+"""
+    ⊕(X::LazySet, v::AbstractVector)
+
+Unicode alias constructor ⊕ (`oplus`) for the lazy translation operator.
+"""
+⊕(X::LazySet, v::AbstractVector) = Translation(X, v)
+
+# ============================ 
+# LazySet interface functions
+# ============================
+
+"""
+    dim(tr::Translation)::Int
+
+Return the dimension of a translation.
+
+### Input
+
+- `tr` -- translation
+
+### Output
+
+The dimension of a translation.
+"""
+function dim(tr::Translation)::Int
+    return length(tr.v)
+end
+
+
+"""
+    σ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+
+Return the support vector of a translation.
+
+### Input
+
+- `d`  -- direction
+- `tr` -- translation
+
+### Output
+
+The support vector in the given direction.
+If the direction has norm zero, the result depends on the wrapped set.
+"""
+function σ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+    return tr.v + σ(d, tr.X)
+end
+
+"""
+    ρ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+
+Return the support function of a translation.
+
+### Input
+
+- `d`  -- direction
+- `tr` -- translation
+
+### Output
+
+The support function in the given direction.
+"""
+function ρ(d::AbstractVector{N}, tr::Translation{N}) where {N<:Real}
+    return dot(d, tr.v) + ρ(d, tr.X)
+end
+
+"""
+    LinearMap(M::AbstractMatrix{N}, tr::Translation{N}) where {N<:Real}
+
+Return the lazy linear map of a translation.
+
+### Input
+
+- `M`  -- matrix
+- `tr` -- translation
+
+### Output
+
+The translation defined by the linear map.
+
+### Notes
+
+This method defines the simplification rule: ``M * (X⊕v) = (M*X) ⊕ (M*v)``,
+returning a new translation.
+"""
+function LinearMap(M::AbstractMatrix{N}, tr::Translation{N}) where {N<:Real}
+    return Translation(M * tr.X, M * tr.v)
+end
+
+"""
+    an_element(tr::Translation)
+
+Return some element of a translation.
+
+### Input
+
+- `tr` -- translation
+
+### Output
+
+An element in the translation.
+
+### Notes
+
+This function first asks for `an_element` function of the wrapped set, then
+translates this element according to the given translation vector. 
+"""
+function an_element(tr::Translation)
+    return an_element(tr.X) + tr.v
+end
+
+"""
+    isempty(tr::Translation)::Bool
+
+Return if a translation is empty or not.
+
+### Input
+
+- `tr` -- translation
+
+### Output
+
+`true` iff the wrapped set is empty.
+"""
+function isempty(tr::Translation)::Bool
+    return isempty(tr.X)
```

Comment:
Wait, is it correct? If `tr.X` is empty but the translation is not the zero vector, we get a singleton. Then:

```julia
function isempty(tr::Translation)::Bool
     return isempty(tr.X) && isempty(tr.v)
end
```
But i don't think that we can create a translation with an empty vector, so `isempty` should be return `false` in all cases?

Reply:
It is correct. The empty set is absorbing for the Minkowski sum:
https://github.com/JuliaReach/LazySets.jl/blob/30422364694ac4610ff3ed0ad80980f4742cd149/src/MinkowskiSum.jl#L46